runtime/lua/vim/version.lua

```diff
@@ -332,6 +336,7 @@ function M.range(spec) -- Adapted from https://github.com/folke/lazy.nvim
         end
       end
     end
+    ---@diagnostic enable: need-check-nil
     return setmetatable({ from = from, to = to }, { __index = VersionRange })
```

Comment:
Again, better to use assert to resolve nil warnings.

Reply:
Refer to my previous response regarding the `from/to` story.